substrate/client/network/src/transport.rs

```diff
@@ -96,7 +96,7 @@ pub fn build_transport(
 	};
 
 	let transport = transport
-		.upgrade(upgrade::Version::V1Lazy)
+		.upgrade(upgrade::Version::V1)
```

Comment:
Note that this is the upgrade version for the transport protocols (e.g. `noise` and `yamux`) not the upgrade version for the higher level protocols, e.g. `libp2p-kad` or `libp2p-request-response`. This will increase the time to establish a connection by 2 round-trips, one for negotiating the security protocol (noise) and one for the multiplexer (yamux).

If I understand https://github.com/paritytech/substrate/pull/14691 correctly you only faced issues with the higher level protocols, not the transport protocols.

Either way, I don't think this has a big impact on Polkadot, as connections are long-lived and thus connection establishment latency is not that important.

Happy to help if you want to dig deeper into the issues you are facing with `V1Lazy`. See also my comment on https://github.com/paritytech/substrate/pull/14691#issuecomment-1675946673.

Reply:
Yes good catch, we don't have to disable `V1Lazy` here. I'll add it back and check if it still works